data/Store.ts

```diff
@@ -692,6 +693,8 @@ export class Store extends HoistBase {
      * for backwards compat with app code predating support for multiple {@link summaryRecords}.
      */
     get summaryRecord(): StoreRecord {
```

Comment:
I was going to update this to `get summaryRecord(): StoreRecord | null {` to reflect this update. However, doing so will make every call to `summaryRecord` require a null check before using the returned records. I didn't want to introduce that level of added complexity before discussing with the team.

Reply:
Yeah - you’ll find a lot of methods / properties in Hoist that can return null. Our base tsconfig does not have strictNullChecks enabled. I think what you have is appropriate since it’s consistent with the rest of the framework.

Reply:
Re tightening down our tsconfig - my goal is for us to (one day) get to setting `noImplicitAny: true`, then we could turn our sights on `strictNullChecks`.  

I do believe that it would be a net positive, but the cost/benefit calculation now is....unclear... given the amount of change it would entail.